### PR TITLE
Remove fori_append in favor of fori_collect

### PR DESCRIPTION
**Pull request by neerajprad**
This removes another control flow primitive that we had accumulated in our collection. :) This isn't needed since `fori_collect` with jax primitives seems to be better or at least competitive on tests and examples that I tried it on. This also remove the `use_prims` arg from `hmc`. Changes are made to the notebook, but I haven't run it so as not to pollute the existing output with runs from a different system.

### CODE REVIEW

**Comment by fehiepsi** [comment]
Thanks for making this PR! I just want to check with you two things:
+ If `progbar=True`, we use slow version (for small models); and if we set it to False, then we use a fast version?
+ We now use slower version (progbar=True) in test?

File: notebooks/logistic_regression.ipynb [hunk 1]
```diff
@@ -154,10 +153,10 @@
    ],
    "source": [
     "start = time.time()\n",
-    "hmc_states = fori_collect(tqdm.trange(num_samples), sample_kernel, hmc_state,\n",
+    "hmc_states = fori_collect(num_samples, sample_kernel, hmc_state,\n",
     "                          transform=lambda state: {\"coefs\": state.z[\"coefs\"],\n",
     "                                                   \"num_steps\": state.num_steps},\n",
-    "                          use_prims=False)\n",
+    "                          progbar=True)\n",
```
Comment (fehiepsi) [inline notebooks/logistic_regression.ipynb]: this won't take effect unless you also make a similar change to `fori_collect` (removing `use_prims` arg)

**Comment by neerajprad** [comment]
If progbar=True, we use slow version (for small models); and if we set it to False, then we use a fast version?

That's right - if the user asks for intermediate diagnostics, we use the slow version, otherwise we aggressively use JIT whenever possible.

We now use slower version (progbar=True) in test?
Why you still keep use_prims from fori_collect?

I completely missed that `fori_collect` also takes in a `progbar` argument in turn. I'll make the change to remove that and restore tests to the faster version.

**Comment by fehiepsi** [comment]
> That's right - if the user asks for intermediate diagnostics, we use the slow version, otherwise we aggressively use JIT whenever possible.

Sound good, totally agree! :D

File: notebooks/logistic_regression.ipynb (hunk 1 above)
Comment (neerajprad) [reply notebooks/logistic_regression.ipynb]: Done.

**Pull request merged**